webui/react/src/routes/utils.ts

```diff
@@ -101,9 +75,14 @@ export const handlePath = (
   }
 };
 
-// Given a react url returns the react route path.
-const getReactPath = (url: string): string => {
-  return parseUrl(url).pathname.replace(process.env.PUBLIC_URL, '');
+// remove host and public_url.
+const stripUrl = (aUrl: string): string => {
+  const url = parseUrl(aUrl);
+  const rest = url.href.replace(url.origin, '');
+  if (rest.startsWith(process.env.PUBLIC_URL)) {
+    return rest.replace(process.env.PUBLIC_URL, '');
+  }
```

Comment:
**question:** if `aUrl` contains the `PUBLIC_URL` won't `parseUrl(aUrl)` have `url.origin` set to `PUBLIC_URL`?

Reply:
If I get the question correctly maybe this is making it more confusing: public_url is named poorly. it's just the subdirectory react is served from and would come in the beginning of `url.path`